service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/utils/XMLSerializer.java

```diff
@@ -0,0 +1,52 @@
+package eu.europeana.cloud.service.dps.storm.utils;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+
+/**
+ * Util class used to serialize object to XML and deserialize XML back to object.
+ *
+ * @param <T> class of the objects
+ */
+public final class XMLSerializer<T> {
+
+    public T deserialize(String xml, Class<T> outputClass) {
```

Comment:
One extra thing, not related directly to this task but fixing it is necessary before closing the user story
inside NodeStatistics we override hashCode method but we didn't check if parentXpath is null before trying to check its hash

Reply:
In fact we do not use null in parentXpath. I have changed that before mainly for cassandra related reason. I will change the constructor to make sure that for null the parentXpath will be an empty string.